### PR TITLE
Support iOS 13 & UIKitForMac

### PR DESCRIPTION
**Pull request by antrix1989**
- Fixed thread checker warnings.
- Provided presentation context for ASWebAuthenticationSession.
- Provide prefersEphemeralWebBrowserSession for ASWebAuthenticationSession.
- Fixed issue when provided view controller was overlapping provided web view.
- Updated test app to compile for UIKitForMac target.

### CODE REVIEW

File: MSAL/src/public/MSALPublicClientApplication.h [hunk 1]
```diff
@@ -257,6 +252,16 @@
...
 - (void)allAccountsFilteredByAuthority:(nonnull MSALAccountsCompletionBlock)completionBlock DEPRECATED_MSG_ATTRIBUTE("Use other synchronous account retrieval API instead.");
 
+/*!
+ Removes all tokens from the cache for this application for the provided account
+ MSAL won't be able to return tokens silently after calling this API, and developer will need to call acquireToken
+ User might need to enter his credentials again after calling this API
+ 
+ @param  account    The account to remove from the cache
+ */
+- (BOOL)removeAccount:(nonnull MSALAccount *)account
```
Comment (oldalton) [inline MSAL/src/public/MSALPublicClientApplication.h]: why did this method get moved?

File: MSAL/test/app/ios/MSALTestAppAcquireTokenViewController.m [hunk 2]
```diff
@@ -106,324 +96,42 @@ - (void)dealloc
...
-}
-
 - (void)viewDidLoad
 {
     [super viewDidLoad];
     
+    self.customWebview = [WKWebView new];
+    [self.wkWebViewContainer addSubview:self.customWebview];
+    self.customWebview.translatesAutoresizingMaskIntoConstraints = NO;
+    [self.customWebview.leftAnchor constraintEqualToAnchor:self.wkWebViewContainer.leftAnchor].active = YES;
```
Comment (oldalton) [inline MSAL/test/app/ios/MSALTestAppAcquireTokenViewController.m]: thanks :)

File: MSAL/test/app/ios/MSALTestAppCacheViewController.m [hunk 3]
```diff
@@ -402,6 +406,91 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
 
 #pragma mark - UITableViewDelegate
 
+- (UISwipeActionsConfiguration *)tableView:(UITableView *)tableView trailingSwipeActionsConfigurationForRowAtIndexPath:(NSIndexPath *)indexPath API_AVAILABLE(ios(11.0))
```
Comment (oldalton) [inline MSAL/test/app/ios/MSALTestAppCacheViewController.m]: can this code be reused?

File: MSAL/test/unit/MSALAcquireTokenTests.m [hunk 4]
```diff
@@ -107,6 +111,12 @@ - (void)setUp
     [self.tokenCache clearWithContext:nil error:nil];
     
     MSIDAADNetworkConfiguration.defaultConfiguration.aadApiVersion = @"v2.0";
+    
+    self.parentController = [UIViewController new];
+    UIView *view = [UIView new];
+    self.window = [UIWindow new];
+    [view setValue:self.window forKey:@"window"];
+    [self.parentController setValue:view forKey:@"view"];
```
Comment (oldalton) [inline MSAL/test/unit/MSALAcquireTokenTests.m]: will this work on macOS?

File: MSAL/MSAL.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -42,7 +42,6 @@
 		04A6B5D2226937890035C7C2 /* MSALRedirectUriVerifier.m in Sources */ = {isa = PBXBuildFile; fileRef = B21E07BC210E5458007E3A3C /* MSALRedirectUriVerifier.m */; };
 		04A6B5D5226937930035C7C2 /* MSALTelemetry.m in Sources */ = {isa = PBXBuildFile; fileRef = 96D9A5471E4AB22900674A85 /* MSALTelemetry.m */; };
 		04A6B5D6226937940035C7C2 /* MSALTelemetry.m in Sources */ = {isa = PBXBuildFile; fileRef = 96D9A5471E4AB22900674A85 /* MSALTelemetry.m */; };
-		04A6B5D7226937980035C7C2 /* (null) in Sources */ = {isa = PBXBuildFile; };
```
Comment (oldalton) [inline MSAL/MSAL.xcodeproj/project.pbxproj]: can you also remove other 3 (null) lines :)?

File: MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h [hunk 6]
```diff
@@ -39,37 +38,37 @@ NS_ASSUME_NONNULL_BEGIN
...
-@property NSArray<MSALAuthority *> *knownAuthorities;
+@property (nonatomic) NSArray<MSALAuthority *> *knownAuthorities;
 
 /*! Enable to return access token with extended lifttime during server outage. */
-@property BOOL extendedLifetimeEnabled;
+@property (nonatomic) BOOL extendedLifetimeEnabled;
 
 /*! List of additional ESTS features that client handles. */
-@property(nullable) NSArray<NSString *> *clientApplicationCapabilities;
+@property (nullable) NSArray<NSString *> *clientApplicationCapabilities;
```
Comment (oldalton) [inline MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h]: nonatomic here as well?

Comment (antrix1989) [reply MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h]: done

File: MSAL/MSAL.xcodeproj/project.pbxproj (hunk 5 above)
Comment (antrix1989) [reply MSAL/MSAL.xcodeproj/project.pbxproj]: done

File: MSAL/src/MSALPublicClientApplication.m [hunk 7]
```diff
@@ -505,13 +503,34 @@ - (void)acquireTokenWithParameters:(MSALInteractiveTokenParameters *)parameters
...
+        {
+            NSError *msidError = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidDeveloperParameter, @"parentViewController is a required parameter on iOS 13.", nil, nil, nil, nil, nil);
+            NSError *msalError = [MSALErrorConverter msalErrorFromMsidError:msidError];
+            completionBlock(nil, msalError);
+            return;
+        }
+        
+        if (parameters.webviewConfig.parentViewController.view.window == nil)
+        {
+            NSError *msidError = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidDeveloperParameter, @"parentViewController has no window!", nil, nil, nil, nil, nil);
```
[the remark this answers is not in the document]
Comment (antrix1989) [reply MSAL/src/MSALPublicClientApplication.m]: done

File: MSAL/test/unit/MSALAcquireTokenTests.m (hunk 4 above)
Comment (antrix1989) [reply MSAL/test/unit/MSALAcquireTokenTests.m]: addressed

**Comment by colinhumber** [comment]
Just a note that the `TARGET_OS_UIKITFORMAC` has been changed to `TARGET_OS_MACCATALYST` in the latest betas.

**Pull request merged**